### DISCUSSION TITLE
How to set separate CSS for a certain type of directory and its subdirectories

### DISCUSSION THREAD

**Discussion by w1ndys** [squidfunk/mkdocs-material#7180] [Q&A] [answered]
For example, I want to set `/path/...` and all directories below it to a color different from the default theme color.

I tried introducing style separately in md, but I think this method is too clumsy (especially when there are many files)

Is there any way to use the line of sight function efficiently?

**Comment by w1ndys** [comment]
> I tried introducing style separately in md, but I think this method is too clumsy (especially when there are many files)

If this is the case, I think that I can improve some efficiency through Python scripts, but once the content needs to be modified, it will require a huge workload.

**Comment by ValiLey** [comment, marked as answer]
Would it be an option to create multiple template files in the override directory (like [main.html](https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks)), each referencing a different stylesheet / different color? Like this:

``` 
{% extends "base.html" %}

{% block extrahead %}
  <link rel="stylesheet" href="{{ 'assets/stylesheets/color-red.css' | url }}">
{% endblock %}
```

You can then assign the templates in the frontmatter of your markdown files. Like this:

```
---
title: I'm a page title
description: I'm a description
template: red.html
---
```
Maybe it's even possible to use the [meta plugin](https://squidfunk.github.io/mkdocs-material/plugins/meta) to assign the template to a whole folder instead of adding it to every single file.

**Reply by w1ndys** [reply]
Thanks for your reply, that sounds great.
Your method is more convenient and efficient than mine
However, this still seems to be a "clumsy way" because it has to write metadata once in each file (free version)
Maybe meta plugin is a good choice.
If you don't choose the meta plug-in, you may use Python and template metadata together, which will be more efficient.

**Reply by ValiLey** [reply]
Here's a workaround from a non-developer (😅) : 
If you use Notepad++ to edit your .md files, you can find and replace the content of multiple files located in a directory. 

Find:
```
---
title:
```

Replace with:
```
---
template: red.html
title:
```

This way you can assign a template to a whole folder without a script.

**Reply by w1ndys** [reply]
sounds good👍🏻
